File: lab1 class/untitled/src/SkinCare.java
public abstract class SkinCare extends Product{
    private String activeComponent;
    protected Boolean remove = false;
    protected SkinCare( String name, String brand, double price, String activeComponent, String scent) {
        super(name, brand, price , scent);
        this.activeComponent = activeComponent;
    }

    public void setRemove(Boolean remove) {
        this.remove = remove;
    }
}

File: lab2_fabrica/fabrica/SkinCare.cs
using System;
using System.Runtime.Serialization;
using CosmeticsBase;
namespace cosmetics
{
    [DataContract]
    [KnownType(typeof(Mask))]
    [KnownType(typeof(Serum))]
    public abstract class SkinCare : Product
    {
        [DataMember]
        protected String activeComponent;
        public String ActiveComponent
        {
            get { return activeComponent; }
            set { activeComponent = value; }
        }
        protected SkinCare(String name, String brand, double price, String activeComponent) : base(name, brand, price)
        {
            this.activeComponent = activeComponent;
        }
    }
}
